### ISSUE TITLE
the reason why 'no detections found by model'?

### ISSUE THREAD

**Issue by 2hyes** [eriklindernoren/PyTorch-YOLOv3#628]
I want to do transfer learning with using Darknet53.conv.74 backbone, coco128 datasets. 

```
---- mAP 0.0
Training Epoch 1: 100%|█████████████████████████| 16/16 [00:10<00:00,  1.59it/s]
Training Epoch 2: 100%|█████████████████████████| 16/16 [00:09<00:00,  1.61it/s]

---- Evaluating Model ----
Detecting objects: 100%|████████████████████████| 16/16 [00:02<00:00,  5.77it/s]
metrics_output:  None
---- mAP not measured (no detections found by model)
```
This is the result of $ python train.py --data_config data/coco.data --pretrained_weights weights/darknet53.conv.74.

Here, why does detection not work? (I didn't change code.)

**Comment by Flova** [comment]
You need more than 2 epochs to train the network on coco. Even if you are using some pre-trained weights in the encoder.

**Comment by 2hyes** [comment]
![스크린샷 2021-02-09 오전 10 25 33](https://user-images.githubusercontent.com/36873744/107302726-26f8bd00-6ac1-11eb-8f68-6a7437fe2621.png)

yes I used 10epochs, and every detection task (except right after 0epoch) does not work. (refer to image)

```
/data/yolov3/coco128/images/train2017/000000000009.jpg
/data/yolov3/coco128/images/train2017/000000000025.jpg
/data/yolov3/coco128/images/train2017/000000000030.jpg
/data/yolov3/coco128/images/train2017/000000000034.jpg
/data/yolov3/coco128/images/train2017/000000000036.jpg
...
```
and my train.txt and valid.txt file is like this.

**Comment by Flova** [comment]
That is indeed weird. Maybe I give it a try in the next days.

**Comment by 2hyes** [comment]
I think because of dataset. I will try training with larger dataset, and let you know soon in here. 
Thank you.

**Comment by Flova** [comment]
But it should easily overfit on this smaller set. Maybe there is something wrong with the formatting and the import.

**Comment by Zzheng-6** [comment]
Did you solve this problem? Because I also encountered this error

**Comment by Flova** [comment]
On the same dataset?

**Comment by 2hyes** [comment]
@Zzheng-6 not yet.

I tried transfer learning with the following conditions.
- dataset: 500 sampled images of COCO2014 dataset
- pretrained weights = darknet53.conv.74
- evaluation_interval = 2

```
| 66    | keyboard       | 0.00000 |
| 67    | cell phone     | 0.00000 |
| 68    | microwave      | 0.00000 |
| 69    | oven           | 0.00000 |
| 70    | toaster        | 0.00000 |
| 71    | sink           | 0.00001 |
| 72    | refrigerator   | 0.00007 |
| 73    | book           | 0.00002 |
| 74    | clock          | 0.00000 |
| 75    | vase           | 0.00000 |
| 76    | scissors       | 0.00000 |
| 77    | teddy bear     | 0.00003 |
| 79    | toothbrush     | 0.00000 |
+-------+----------------+---------+
---- mAP 2.601784971674012e-05

Training Epoch 1: 100%|██████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.62it/s]
---- training loss 341.9720458984375

Training Epoch 2: 100%|██████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.51it/s]
---- training loss 304.5881652832031


---- Evaluating Model ----
Detecting objects:   0%|                                                                        | 0/63 [00:00<?, ?it/s]
```

Result is like this. 
First detecting task is working, but the second detecting task is not working and just disconnects the server I use.

**Comment by Zzheng-6** [comment]
That's really a headache.
Thank you for your reply, can you tell me if you find a solution?

------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "eriklindernoren/PyTorch-YOLOv3"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2021年2月14日(星期天) 下午5:34
收件人:&nbsp;"eriklindernoren/PyTorch-YOLOv3"<PyTorch-YOLOv3@noreply.github.com&gt;;
抄送:&nbsp;"两颗西柚"<1349478794@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
主题:&nbsp;Re: [eriklindernoren/PyTorch-YOLOv3] the reason why 'no detections found by model'? (#628)





 
@Zzheng-6 not yet.
 
I tried transfer learning with the following conditions.
  
dataset: 500 sampled images of COCO2014 dataset
 
pretrained weights = darknet53.conv.74
 
evaluation_interval = 2
  | 66    | keyboard       | 0.00000 | | 67    | cell phone     | 0.00000 | | 68    | microwave      | 0.00000 | | 69    | oven           | 0.00000 | | 70    | toaster        | 0.00000 | | 71    | sink           | 0.00001 | | 72    | refrigerator   | 0.00007 | | 73    | book           | 0.00002 | | 74    | clock          | 0.00000 | | 75    | vase           | 0.00000 | | 76    | scissors       | 0.00000 | | 77    | teddy bear     | 0.00003 | | 79    | toothbrush     | 0.00000 | +-------+----------------+---------+ ---- mAP 2.601784971674012e-05 Training Epoch 1: 100%|██████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.62it/s] ---- training loss 341.9720458984375 Training Epoch 2: 100%|██████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.51it/s] ---- training loss 304.5881652832031 ---- Evaluating Model ---- Detecting objects:   0%|                                                                        | 0/63 [00:00<?, ?it/s]  
Result is like this.
 First detecting task is working, but the second detecting task is not working and just disconnects the server I use.

**Comment by Dan944** [comment]
I met the same problem, it seems that all the conf are too low, which makes that the map is quite low even the test set are train set

**Comment by cesarhcq** [comment]
> @Zzheng-6 not yet.
> 
> I tried transfer learning with the following conditions.
> 
> * dataset: 500 sampled images of COCO2014 dataset
> * pretrained weights = darknet53.conv.74
> * evaluation_interval = 2
> 
> ```
> | 66    | keyboard       | 0.00000 |
> | 67    | cell phone     | 0.00000 |
> | 68    | microwave      | 0.00000 |
> | 69    | oven           | 0.00000 |
> | 70    | toaster        | 0.00000 |
> | 71    | sink           | 0.00001 |
> | 72    | refrigerator   | 0.00007 |
> | 73    | book           | 0.00002 |
> | 74    | clock          | 0.00000 |
> | 75    | vase           | 0.00000 |
> | 76    | scissors       | 0.00000 |
> | 77    | teddy bear     | 0.00003 |
> | 79    | toothbrush     | 0.00000 |
> +-------+----------------+---------+
> ---- mAP 2.601784971674012e-05
> 
> Training Epoch 1: 100%|██████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.62it/s]
> ---- training loss 341.9720458984375
> 
> Training Epoch 2: 100%|██████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.51it/s]
> ---- training loss 304.5881652832031
> 
> 
> ---- Evaluating Model ----
> Detecting objects:   0%|                                                                        | 0/63 [00:00<?, ?it/s]
> ```
> 
> Result is like this.
> First detecting task is working, but the second detecting task is not working and just disconnects the server I use.

What is the image size which are you using?

**Comment by 2hyes** [comment]
I use 'image_size = 416'.

**Comment by Flova** [comment]
I tried to train it with the coco128 dataset, but it takes significantly longer than 10 Epochs to make useful detections. Because of the low object probability for a given cell, it's best for the network to predict nothing in the beginning. I also updated the loss function. Maybe try again.

**Comment by Flova** [comment]
> @Zzheng-6 not yet.
> 
> I tried transfer learning with the following conditions.
> 
>     * dataset: 500 sampled images of COCO2014 dataset
> 
>     * pretrained weights = darknet53.conv.74
> 
>     * evaluation_interval = 2
> 
> 
> ```
> | 66    | keyboard       | 0.00000 |
> | 67    | cell phone     | 0.00000 |
> | 68    | microwave      | 0.00000 |
> | 69    | oven           | 0.00000 |
> | 70    | toaster        | 0.00000 |
> | 71    | sink           | 0.00001 |
> | 72    | refrigerator   | 0.00007 |
> | 73    | book           | 0.00002 |
> | 74    | clock          | 0.00000 |
> | 75    | vase           | 0.00000 |
> | 76    | scissors       | 0.00000 |
> | 77    | teddy bear     | 0.00003 |
> | 79    | toothbrush     | 0.00000 |
> +-------+----------------+---------+
> ---- mAP 2.601784971674012e-05
> 
> Training Epoch 1: 100%|██████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.62it/s]
> ---- training loss 341.9720458984375
> 
> Training Epoch 2: 100%|██████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.51it/s]
> ---- training loss 304.5881652832031
> 
> 
> ---- Evaluating Model ----
> Detecting objects:   0%|                                                                        | 0/63 [00:00<?, ?it/s]
> ```
> 
> Result is like this.
> First detecting task is working, but the second detecting task is not working and just disconnects the server I use.

This looks like an issue due to the evaluation. During training, no non-maximum suppression is applied. Evaluation on the other hand uses nms. The nms scales not well for large quantities of predicted objects. So if the network predicts many wrong objects in the beginning it is able to get "stuck" in the evaluation. I lowered the timeout for the nms estimation because a wrong nms doesn't matter that much in a situation where we have unrealistic amounts of objects.
